cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlService.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.services;
+
+import com.google.inject.Inject;
+import io.cdap.cdap.common.BadRequestException;
+import io.cdap.cdap.proto.id.NamespaceId;
+import io.cdap.cdap.proto.security.StandardPermission;
+import io.cdap.cdap.proto.sourcecontrol.RepositoryConfig;
+import io.cdap.cdap.security.spi.authentication.AuthenticationContext;
+import io.cdap.cdap.security.spi.authorization.AccessEnforcer;
+import io.cdap.cdap.spi.data.StructuredTableContext;
+import io.cdap.cdap.spi.data.TableNotFoundException;
+import io.cdap.cdap.spi.data.transaction.TransactionRunner;
+import io.cdap.cdap.spi.data.transaction.TransactionRunners;
+import io.cdap.cdap.store.RepositoryTable;
+
+/**
+ * Service that manages source control for .
+ */
+public class SourceControlService {
+
+  private final AccessEnforcer accessEnforcer;
+  private final AuthenticationContext authenticationContext;
+  private final TransactionRunner transactionRunner;
+
+  @Inject
+  public SourceControlService(TransactionRunner transactionRunner,
```

Comment:
We would want `SourceControlService` to also provide higher order functionalities over basic git operations and use in taskWorker. Not sure if injecting TransactionRunner here makes sense in that case. 
If we want to put db transaction logic in SourceControlService we need to add some documentation to make it clear we cannot use corresponding functions inside taskWorker

Reply:
I guess the one you mentioned is `SourceControlManager`? This one is following the current CDAP archietecture pattern: handler layer --> service layer --> store layer. It's a downstream layer of httpHandler that deals with business logic, and it controls when and where the transaction should happen/close, also resonating this comment: https://github.com/cdapio/cdap/pull/14838/files#r1085773177